packages/sanity/src/core/form/inputs/ReferenceInput/PreviewReferenceValue.tsx

```diff
@@ -57,21 +59,91 @@ export function PreviewReferenceValue(props: {
     }
   }
 
+  const availability = referenceInfo.result.availability
+
+  const notFound = availability.reason === 'NOT_FOUND'
+  const insufficientPermissions = availability.reason === 'PERMISSION_DENIED'
+  if (insufficientPermissions || notFound) {
+    return (
+      <Inline space={2}>
+        <Box padding={1}>
+          <Flex align="center">
+            <Box flex={1} paddingY={2}>
+              <Text muted>Document unavailable</Text>
+            </Box>
+          </Flex>
+        </Box>
+        {insufficientPermissions || notFound ? (
+          <Box>
+            <Tooltip
+              portal
+              content={
+                notFound ? (
+                  <UnavailableMessage title="Not found" icon={HelpCircleIcon}>
+                    The referenced document does not exist
+                    <br />
+                    (id: <code>{value._ref}</code>)
+                  </UnavailableMessage>
+                ) : (
+                  <UnavailableMessage title="Insufficcient permissions" icon={AccessDeniedIcon}>
+                    The referenced document could not be accessed due to insufficient permissions
+                  </UnavailableMessage>
+                )
+              }
+            >
+              <TextWithTone tone="default">
+                <HelpCircleIcon />
+              </TextWithTone>
+            </Tooltip>
+          </Box>
+        ) : null}
+      </Inline>
+    )
+  }
+
   const refTypeName = referenceInfo.result?.type
   const refType = type.to.find((toType) => toType.name === refTypeName)
 
   if (!refType) {
     return (
-      <Stack space={2} padding={2}>
-        The referenced document is of invalid type: ({refTypeName || 'unknown'})
-        <pre>{JSON.stringify(value, null, 2)}</pre>
-      </Stack>
+      <Inline space={2}>
+        <Box padding={1}>
+          <Flex align="center">
+            <Box flex={1} paddingY={2}>
+              <Text muted>Type of referenced document not declared in schema</Text>
+            </Box>
+          </Flex>
+        </Box>
+        <Box>
+          <Tooltip
+            portal
+            content={
+              <Stack space={3} padding={3}>
+                <Text size={1}>
+                  Referenced document (<code>{value._ref}</code>) is of type{' '}
+                  <code>{referenceInfo.result.type}</code>.
+                </Text>
+                <Text size={1}>
+                  According to the schema, referenced documents can only be of type{' '}
+                  {humanizeList(
+                    type.to.map((toType) => <code key={toType.name}>{toType.name}</code>),
+                    'or'
+                  )}
+                </Text>
+              </Stack>
+            }
+          >
+            <TextWithTone tone="default">
+              <HelpCircleIcon />
+            </TextWithTone>
+          </Tooltip>
+        </Box>
+      </Inline>
```

Comment:
```suggestion
      <Flex align="center" justify="flex-start">
        <Box padding={1}>
          <Flex align="center">
            <Box flex={1} paddingY={2}>
              <Text muted>Type of referenced document not declared in schema</Text>
            </Box>
          </Flex>
        </Box>
        <Box>
          <Tooltip
            portal
            content={
              <Stack space={3} padding={3}>
                <Text size={1}>
                  Referenced document (<code>{value._ref}</code>) is of type{' '}
                  <code>{referenceInfo.result.type}</code>.
                </Text>
                <Text size={1}>
                  According to the schema, referenced documents can only be of type{' '}
                  {humanizeList(
                    type.to.map((toType) => <code key={toType.name}>{toType.name}</code>),
                    'or'
                  )}
                  .
                </Text>
              </Stack>
            }
          >
            <Box padding={2}>
              <TextWithTone tone="default">
                <HelpCircleIcon />
              </TextWithTone>
            </Box>
          </Tooltip>
        </Box>
      </Flex>
```

**suggestion**: Using a wrapping `<Flex>` rather than `<Inline>` to ensure that the icon / context menu wraps gracefully at smaller breakpoints, e.g.

<img width="379" alt="image" src="https://github.com/sanity-io/sanity/assets/209129/fc82605f-5f09-437c-9c98-dec39de4cc15">

vs.

<img width="381" alt="image" src="https://github.com/sanity-io/sanity/assets/209129/8c6bbde0-7d18-4299-bcb0-708569be461f">

Reply:
@robinpyon which one is 'before' and which one is 'after' your change?
I guess that the icon should *not* wrap under the text?

Reply:
That's correct – first image is 'after', this could have been clearer!